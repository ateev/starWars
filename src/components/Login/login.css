.login-container{
  width: 50%;
  margin: 0 auto;
  margin-top: 40px;
}
.login-container input {
  padding: 5px 15px;
  margin: 0 auto;
  margin-right: 20px;
  margin-bottom: 20px;
}
input[type="submit"] {
  border-radius: 4px;
  border: 0;
  width: 100px;
  height: 30px;
  background: #222222;
  color: #fff;
  cursor: pointer;
}
.form-title {
  padding: 40px;
  color: #222222;
  font-size: 20px;
}