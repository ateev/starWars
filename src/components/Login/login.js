import React, { Component } from 'react';
import './login.css';

class LoginContainer extends Component {
  constructor() {
    super();
    this.state = {
      username: 'Luke Skywalker',
      password: '19BBY',
    }
  }
  handleChange = (ev) => {
    const newObj = {};
    newObj[ev.target.name] = ev.target.value;
    this.setState(newObj);
  }
  submitForm = (ev) => {
    ev.preventDefault();
    this.props.submitCallback(this.state.username, this.state.password);
  }
  render() {
    return (
      <div className="login-container">
        <form onSubmit={ this.submitForm }>
          <input
            type="text"
            name="username"
            value={this.state.username}
            onChange={this.handleChange}
          />
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={this.handleChange}
          />
          <input type="submit" value="Login" />
        </form>
      </div>
    );
  }
}

export default LoginContainer;
