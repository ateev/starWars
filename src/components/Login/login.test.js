import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Login from './Login.js';

describe('<Login />', () => {
    it('renders without crashing', () => {
        const component = shallow(
            <Login />
        );
        expect(component).toMatchSnapshot();
    });
})
