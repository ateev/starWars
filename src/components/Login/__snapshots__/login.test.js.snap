exports[`<Login /> renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="login-container">
    <form
        onSubmit={[Function]}>
        <input
            name="username"
            onChange={[Function]}
            type="text"
            value="Luke Skywalker" />
        <input
            name="password"
            onChange={[Function]}
            type="password"
            value="19BBY" />
        <input
            type="submit"
            value="Login" />
    </form>
</div>,
  "nodes": Array [
    <div
      className="login-container">
      <form
            onSubmit={[Function]}>
            <input
                  name="username"
                  onChange={[Function]}
                  type="text"
                  value="Luke Skywalker" />
            <input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value="19BBY" />
            <input
                  type="submit"
                  value="Login" />
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LoginContainer />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LoginContainer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "password": "19BBY",
          "username": "Luke Skywalker",
        },
        "submitForm": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="login-container">
          <form
                    onSubmit={[Function]}>
                    <input
                              name="username"
                              onChange={[Function]}
                              type="text"
                              value="Luke Skywalker" />
                    <input
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value="19BBY" />
                    <input
                              type="submit"
                              value="Login" />
          </form>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="login-container">
          <form
                    onSubmit={[Function]}>
                    <input
                              name="username"
                              onChange={[Function]}
                              type="text"
                              value="Luke Skywalker" />
                    <input
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value="19BBY" />
                    <input
                              type="submit"
                              value="Login" />
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LoginContainer />,
}
`;
